src/mca/plm/base/plm_base_jobid.c

```diff
@@ -140,11 +141,22 @@ int prte_plm_base_create_jobid(prte_job_t *jdata)
     PMIX_LOAD_NSPACE(jdata->nspace, tmp);
     free(tmp);
     // This routine will create a new prte_job_t structure and add it to the
-    // prte_job_data hash table at the ".N" position.
+    // prte_job_data hash table at the ".N" position. Note that that 'new'
+    // object is not the 'jdata' referenced here. After this call we have
+    // two objects in the system with the same jobid/nspace.
     PRTE_PMIX_CONVERT_NSPACE(rc, &jdata->jobid, jdata->nspace);
     if (PRTE_SUCCESS != rc) {
         return rc;
     }
+    // Make sure the object in this hash table is the one passed to this function.
```

Comment:
This comment doesn't seem to match whet the following code does, releasing the old structure

Reply:
I cleaned up the comment a bit.